src/NeoSharp.BinarySerialization/BinaryDeserializer.cs

```diff
@@ -0,0 +1,190 @@
+using NeoSharp.BinarySerialization.Cache;
+using NeoSharp.BinarySerialization.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace NeoSharp.BinarySerialization
+{
+    public class BinaryDeserializer : IBinaryDeserializer
```

Comment:
I prefer to keep serializer and deserializer in a single class/interface because it is very unusual to split them. We can run into the situation where the discrepancy happens.

Reply:
All serializers like Json, have both classes, this is the reason for my "split"

Reply:
I can create a IBinaryConverter with both implementations, what do you think?

Reply:
You can define interface like this but for actual class it is a lot of delegation work. My vote is against.

Reply:
I think that all of this have three, one for all (made mor simple) and the other (serializer and deserializer)